entity-server-api/src/main/java/org/terracotta/entity/PassiveServerEntity.java

```diff
@@ -16,46 +16,56 @@
- *  connected clients, nor can it return data to them.
- * Note that passive entities have no invoke response but we still accept the generic R so that accessors of the codec will
- * see a consistent type landscape.
+ * <p>The methods specifically supported by passive entities.  Note that a passive doesn't know anything about
+ *  connected clients, nor can it return data to them.</p>
+ * <p>Note that passive entities have no invoke response but we still accept the generic R so that accessors of the codec
+ *  will see a consistent type landscape.</p>
  */
 public interface PassiveServerEntity<M extends EntityMessage, R extends EntityResponse> extends CommonServerEntity<M, R> {
   /**
-   * Invoke a call on the given entity.  Note that passive entities can't return data to the client.
-   * This method is called both in the cases of normal client->server method invocation but also in the case of
-   * server->server passive synchronization.
+   * <p>Invoke a call on the given entity.  Note that passive entities can't return data to the client.</p>
+   * <p>This method is called both in the cases of normal client->server method invocation but also in the case of
+   *  server->server passive synchronization.</p>
+   * <p>Note that the thread used to make this call is determined by consulting the entity's ConcurrencyStrategy so it may be
+   *  called concurrently with other invokes.</p>
+   * <p>A note about passive synchronization messages:  When the active is given the opportunity to synchronize the data
+   *  under a given key, the messages it creates and sends for synchronization will be executed in that key, on the passive,
+   *  via this invoke() entry-point.</p>
    *
    * @param message The message from a client or upstream active server
    */
   void invoke(M message);
 
   /**
-   * Called on MANAGEMENT_KEY to notify the receiver that it is about to start receiving synchronization messages.
+   * <p>Called on {@link ConcurrencyStrategy#MANAGEMENT_KEY} to notify the receiver that it is about to start receiving
+   *  synchronization messages.</p>
+   * <p>Note that this is called after createNew().</p>
    */
   void startSyncEntity();
 
   /**
-   * Called on MANAGEMENT_KEY to notify the receiver synchronization is complete and it will see no more synchronization
-   * messages.
+   * Called on {@link ConcurrencyStrategy#MANAGEMENT_KEY} to notify the receiver synchronization is complete and it will see
+   *  no more synchronization messages.
    */
   void endSyncEntity();
 
   /**
-   * Called on concurrencyKey to notify the receiver that it is about to start receiving synchronization messages on this
-   * key.
+   * <p>Called on concurrencyKey to notify the receiver that it is about to start receiving synchronization messages on this
+   *  key.</p>
+   * <p>To clarify:  the concurrencyKey is both the argument (so the entity can logically act on it) but is also the key
+   *  where this call is executed.</p>
```

Comment:
Current behavior is to iterate sync in order and serially over the keys as presented by the entity implementor.  This is to minimize the impact of sync on client throughput.  The question is do we want to lock ourselves into this behavior.